src/status_im/ui/screens/chat/pinned_messages.cljs

```diff
@@ -0,0 +1,126 @@
+            [status-im.utils.platform :as platform]
+            [quo.react :as quo.react]
+            [status-im.ui.components.topbar :as topbar]))
+
+(defn pins-topbar []
+  (let [{:keys [group-chat chat-id chat-name]}
+        @(re-frame/subscribe [:chats/current-chat])
+        pinned-messages @(re-frame/subscribe [:chats/pinned chat-id])
+        [first-name _] @(re-frame.core/subscribe [:contacts/contact-two-names-by-identity chat-id])]
+    [topbar/topbar {:show-border? true
+                    :title        (if group-chat chat-name first-name)
+                    :subtitle     (if (= (count pinned-messages) 0)
+                                    (i18n/label :t/no-pinned-messages)
+                                    (i18n/label-pluralize (count pinned-messages) :t/pinned-messages-count))}]))
+
+(defn get-space-keeper-ios [bottom-space panel-space active-panel text-input-ref]
+  (fn [state]
+    ;; NOTE: Only iOS now because we use soft input resize screen on android
+    (when platform/ios?
+      (cond
+        (and state
+             (< @bottom-space @panel-space)
+             (not @active-panel))
+        (reset! bottom-space @panel-space)
+
+        (and (not state)
+             (< @panel-space @bottom-space))
+        (do
+          (some-> ^js (quo.react/current-ref text-input-ref) .focus)
+          (reset! panel-space @bottom-space)
+          (reset! bottom-space 0))))))
+
+(defn pinned-messages-empty []
+  [react/view {:style {:flex 1
+                       :align-items :center
+                       :justify-content :center}}
+   [react/text {:style style/intro-header-description}
+    (i18n/label :t/pinned-messages-empty)]])
+
+(defonce messages-list-ref (atom nil))
+
+(defn render-pin-fn [{:keys [outgoing type] :as message}
+                     idx
+                     _
+                     {:keys [group-chat public? current-public-key space-keeper chat-id show-input?]}]
+  [react/view
+   (if (= type :datemark)
+     [message-datemark/chat-datemark (:value message)]
+     (if (= type :gap)
+       [gap/gap message idx messages-list-ref false chat-id]
+       ; message content
+       [message/chat-message
+        (assoc message
+               :incoming-group (and group-chat (not outgoing))
+               :group-chat group-chat
+               :public? public?
+               :current-public-key current-public-key
+               :show-input? show-input?
+               :pinned true)
+        space-keeper]))])
+
+(def list-key-fn #(or (:message-id %) (:value %)))
+(def list-ref #(reset! messages-list-ref %))
+
+(defn pinned-messages-view [{:keys [chat bottom-space space-keeper show-input?]}]
+  (let [{:keys [group-chat chat-id public?]} chat
+        pinned-messages @(re-frame/subscribe [:chats/pinned-messages-stream chat-id])
+        current-public-key @(re-frame/subscribe [:multiaccount/public-key])]
+    ;;do not use anonymous functions for handlers
```

Comment:
Why?

Reply:
because each time new function will be generated, and it will be new object for react, but it should be the same function